### PR TITLE
fix(zimbra): #MC-272 add unit tests

### PR DESCRIPTION
**Pull request by alicedraillard**
## Describe your changes
Unit tests to secure helpers

## Issue ticket number and link
[MC-272](https://jira.support-ent.fr/browse/MC-272)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by qpic-cgi — APPROVED** [review]

File: src/test/java/net/atos/entng/calendar/helper/CalendarHelperTest.java [hunk 1]
```diff
@@ -1,7 +1,11 @@
 package net.atos.entng.calendar.helper;
 
+import com.redis.M;
+import com.redis.S;
+import com.redis.U;
 import fr.wseduc.webutils.I18n;
 import fr.wseduc.webutils.eventbus.ResultMessage;
+import io.advantageous.boon.core.Str;
```
Comment (BastienBenard) [inline src/test/java/net/atos/entng/calendar/helper/CalendarHelperTest.java:8]: Ils sont utiles ces import ?

File: src/test/java/net/atos/entng/calendar/helper/CalendarHelperTest.java [hunk 2]
```diff
@@ -44,14 +49,17 @@ public class CalendarHelperTest {
 
     private EventServiceMongoImpl eventServiceMongoImpl;
     private CalendarHelper calendarHelper;
+    private CalendarModel calendarModel;
 
     @Before
     public void setUp(TestContext context) throws Exception {
         PowerMockito.spy(EventServiceMongoImpl.class);
         PowerMockito.spy(CalendarHelper.class);
+        PowerMockito.spy(CalendarModel.class);
```
Comment (BastienBenard) [inline src/test/java/net/atos/entng/calendar/helper/CalendarHelperTest.java:58]: Je ne pense pas que cela te sert

File: src/main/java/net/atos/entng/calendar/controllers/CalendarController.java [hunk 3]
```diff
@@ -82,7 +82,7 @@ public CalendarController(String collection, ServiceFactory serviceFactory, Even
         final EventStore eventStore = EventStoreFactory.getFactory().getEventStore(Calendar.class.getSimpleName());
         this.eventHelper = new org.entcore.common.events.EventHelper(eventStore);
         this.calendarHelper = new CalendarHelper(collection, serviceFactory, eb, config);
-        this.platformHelper = new PlatformHelper(collection, serviceFactory);
+        this.platformHelper = new PlatformHelper(serviceFactory);
```
Comment (BastienBenard) [inline src/main/java/net/atos/entng/calendar/controllers/CalendarController.java:85]: Si tu ajoute juste des tests unitaires pourquoi changé le code ?

File: src/main/java/net/atos/entng/calendar/helpers/PlatformHelper.java [hunk 4]
```diff
@@ -21,15 +21,16 @@
...
     protected static final Logger log = LoggerFactory.getLogger(PlatformHelper.class);
 
     private final PlatformService platformService;
+    private final JsonObject config;
 
-    public PlatformHelper(String collection, ServiceFactory serviceFactory) {
-        super(collection, null);
+    public PlatformHelper(ServiceFactory serviceFactory) {
         this.platformService = serviceFactory.platformService();
+        this.config = serviceFactory.getConfig();
```
Comment (BastienBenard) [inline src/main/java/net/atos/entng/calendar/helpers/PlatformHelper.java:33]: same

**Review by sjollois — APPROVED** [review]

File: src/main/java/net/atos/entng/calendar/controllers/CalendarController.java (hunk 3 above)
Comment (alicedraillard) [reply src/main/java/net/atos/entng/calendar/controllers/CalendarController.java:85]: non utilisé

File: src/main/java/net/atos/entng/calendar/helpers/PlatformHelper.java (hunk 4 above)
Comment (alicedraillard) [reply src/main/java/net/atos/entng/calendar/helpers/PlatformHelper.java:33]: j'ai ai besoin pour pouvoir mocker la config

**Pull request merged**